infrastructure/src/test/java/org/corfudb/infrastructure/management/failuredetector/HealingAgentTest.java

```diff
@@ -2,79 +2,117 @@
 
 import org.corfudb.infrastructure.NodeNames;
 import org.corfudb.infrastructure.RemoteMonitoringService.DetectorTask;
-import org.corfudb.infrastructure.management.ClusterAdvisor;
+import org.corfudb.infrastructure.datastore.DataStore;
 import org.corfudb.infrastructure.management.PollReport;
 import org.corfudb.protocols.wireprotocol.ClusterState;
-import org.corfudb.protocols.wireprotocol.failuredetector.NodeRank;
+import org.corfudb.protocols.wireprotocol.failuredetector.FileSystemStats;
 import org.corfudb.runtime.view.Layout;
-import org.corfudb.util.concurrent.SingletonResource;
 import org.junit.jupiter.api.Test;
 
 import java.util.Optional;
 import java.util.concurrent.CompletableFuture;
-import java.util.concurrent.ExecutorService;
 
+import static org.corfudb.infrastructure.management.NodeStateTestUtil.nodeState;
+import static org.corfudb.protocols.wireprotocol.failuredetector.NodeConnectivity.ConnectionStatus.OK;
 import static org.junit.jupiter.api.Assertions.assertEquals;
-import static org.mockito.ArgumentMatchers.any;
 import static org.mockito.ArgumentMatchers.anySet;
-import static org.mockito.ArgumentMatchers.isA;
+import static org.mockito.Matchers.same;
 import static org.mockito.Mockito.doReturn;
 import static org.mockito.Mockito.mock;
-import static org.mockito.Mockito.spy;
 import static org.mockito.Mockito.times;
 import static org.mockito.Mockito.verify;
-import static org.mockito.Mockito.when;
 
-class HealingAgentTest {
+public class HealingAgentTest {
 
     @Test
-    void detectHealingNoFailedNodes() {
-        ClusterAdvisor adviserMock = mock(ClusterAdvisor.class);
+    void unsuccessfulHealingOfAFailedNode() {
+        final String localEndpoint = NodeNames.C;
+
+        FailureDetectorTestContext ctx = new FailureDetectorTestContext(localEndpoint);
+
+        //an example how "setup" function can be used to customize FD test context
+        ctx.setupDataStore(fsDs -> {
+            DataStore dsMock = mock(DataStore.class);
+            fsDs.dataStore(dsMock);
+        });
+
+        ctx.setupClusterState(clusterStateCtx -> {
+            clusterStateCtx.unresponsiveNodes(ctx.localEndpoint);
+
+            final long epoch = 0;
+            FileSystemStats fsStats = clusterStateCtx.getFsStats();
+            clusterStateCtx.setupNodes(
+                    nodeState(NodeNames.A, epoch, Optional.of(fsStats), OK, OK, OK),
+                    nodeState(NodeNames.B, epoch, Optional.of(fsStats), OK, OK, OK),
+                    nodeState(NodeNames.C, epoch, Optional.of(fsStats), OK, OK, OK)
```

Comment:
`localEndpoint` instead of `NodeNames.C`?

Reply:
done